hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/hedera/utils/AddressBookUtils.java

```diff
@@ -136,4 +139,28 @@ public static NodeMetadata classicMetadataFor(
     public static Stream<Long> nodeIdsFrom(AddressBook addressBook) {
         return stream(addressBook.spliterator(), false).map(Address::getNodeId).map(NodeId::id);
     }
+
+    /**
+     *  Returns service end point base on the host and port.
+     *
+     * @param host is an ip or domain name
+     * @param port
+     * @return ServiceEndpoint
+     */
+    public static ServiceEndpoint endpointFor(@NonNull final String host, final int port) {
+        final Pattern IPV4_ADDRESS_PATTERN = Pattern.compile("^((25[0-5]|(2[0-4]|1\\d|[1-9]|)\\d)\\.?\\b){4}$");
+        final var builder = ServiceEndpoint.newBuilder().setPort(port);
+        if (IPV4_ADDRESS_PATTERN.matcher(host).matches()) {
+            final var octets = host.split("[.]");
+            builder.setIpAddressV4(ByteString.copyFrom((new byte[] {
+                (byte) Integer.parseInt(octets[0]),
```

Comment:
I noticed this (byte)int call actually loss some important info during test. If you pass "350.1.1.1", you will still get the correct value. I used ByteString.copyFromUtf8() instead to test the error case.

Reply:
Should add a comment here saying not to use for testing invalid ServiceEndpoint.